forte/data/data_store.py

```diff
@@ -243,14 +271,33 @@ def _new_group(self, type_name: str, member_type: str) -> List:
         """
 
         tid: int = self._new_tid()
+
+        self._add_new_type(type_name)
         entry = [member_type, [], tid, type_name]
-        entry += len(self._type_attributes[type_name]) * [None]
+        try:
+            entry += len(self._type_attributes[type_name]) * [None]
+        except KeyError as e:
+            raise KeyError(
+                f"Group type with id {type_name} does not exist."
+            ) from e
         return entry
```

Comment:
I notice the same lines of code appear multiple times here in all the `_new_xx` function, I think we should merge them together.